zerver/lib/retention.py

```diff
@@ -301,7 +300,7 @@ def archive_stream_messages(realm: Realm, chunk_size: int=MESSAGE_BATCH_SIZE) ->
 def archive_messages(chunk_size: int=MESSAGE_BATCH_SIZE) -> None:
     logger.info("Starting the archiving process with chunk_size {}".format(chunk_size))
 
-    for realm in Realm.objects.all():
+    for realm in Realm.objects.exclude(string_id='zulipinternal'):
```

Comment:
Do we need the `exclude` here?  I would have thought it was unnecessary because the `message_retention_days` for the system bot realm would be unset.

Reply:
Just wanted to check in on this.

Reply:
Indeed we don't need the ``exclude``, but I thought it'd make sense here to make it clear we skip that realm instead of relying implicitly on the reader knowing ``message_retention_days`` is not set for the internal realm.

Reply:
That seems like reasonable defensive code to me, and it shouldn't slow anything down, really.

Reply:
Though at least it should be better to use ``settings.SYSTEM_BOT_REALM`` instead of hard-coded ``zulipinternal`` so I changed that for now.